src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -480,3 +480,87 @@
       (filter (fn [{:keys [tokens]}]
                 (some positive-balance-in-any-chain? tokens))
               operable-account))))
+
+(defn send-details-map
+  "Generates a new map with all the sen details using the original
+  details we've received from `status-go`. The new one contains only
+  the keys we need."
+  [send-details]
+  (let [{:keys [uuid
+                sendType
+                fromAddress
+                toAddress
+                fromChain
+                toChain
+                fromAmount
+                toAmount
+                fromAsset
+                toAsset
+                username
+                publicKey
+                packId]} send-details]
+    {:uuid         uuid
+     :send-type    sendType
+     :address-from fromAddress
+     :address-to   toAddress
+     :tx-to        ""
+     :from-chain   fromChain
+     :to-chain     toChain
+     :from-amount  fromAmount
+     :to-amount    toAmount
+     :from-asset   fromAsset
+     :to-asset     toAsset
+     :username     username
+     :public-key   publicKey
+     :pack-id      packId
+     :tx-hash      ""
+     :approval-tx? false}))
+
+(defn details-map
+  "Generates map with all the transaction details based on what we've
+  got from status-go as `send-details` and `sent-transaction`."
+  [send-details sent-transaction]
+  (let [send-details         (send-details-map send-details)
+        {:keys [toAddress fromChain toChain amountIn
+                amountOut fromToken toToken hash
+                approvalTx]} sent-transaction
+        amount-in            (money/from-hex amountIn)
+        amount-out           (money/from-hex amountOut)
+        sent-transaction?    (and sent-transaction (> (-> sent-transaction :hash count) 0))]
+    (if sent-transaction?
+      (cond-> send-details
+        true                    (assoc :tx-to toAddress)
+        (> fromChain 0)         (assoc :from-chain fromChain)
+        (> toChain 0)           (assoc :to-chain toChain)
+        (not= amount-in "0")    (assoc :from-amount amount-in)
+        (not= amount-out "0")   (assoc :to-amount amount-out)
+        (> (count fromToken) 0) (assoc :from-asset fromToken)
+        (> (count toToken) 0)   (assoc :to-asset toToken)
+        true                    (assoc :tx-hash hash)
+        true                    (assoc :approval-tx? approvalTx))
+      send-details)))
+
+(defn contact-name-by-address
+  [db address]
+  (or (get-in db [:wallet :accounts address :name])
+      (get-in db [:contacts/contacts address :primary-name])))
```

Comment:
Quick question: The contact address is different from the wallet address. How will this work?

Reply:
we simply try to get either one or another